hail/python/test/hailtop/aiotools/test_time_limited_max_size_cache.py

```diff
@@ -0,0 +1,146 @@
+import pytest
+import asyncio
+
+from hailtop.aiotools.time_limited_max_size_cache import TimeLimitedMaxSizeCache
+
+
+pytestmark = pytest.mark.asyncio
+
+
+one_second_ns = 1000 * 1000 * 1000
+one_day_ns = 24 * 60 * 60 * one_second_ns
+
+
+async def test_simple():
+    async def load_secret(k: int):
+        return k ** 2
+
+    c = TimeLimitedMaxSizeCache(load_secret, one_day_ns, 2)
+    assert await c.lookup(3) == 9
+    assert await c.lookup(10) == 100
+    assert await c.lookup(3) == 9
+    assert await c.lookup(3) == 9
+    assert await c.lookup(10) == 100
+    assert await c.lookup(4) == 16
+
+
+async def test_max_size():
+    load_counts = 0
+    async def load_secret_only_twice(k: int):
+        nonlocal load_counts
+        if load_counts == 2:
+            raise ValueError("already loaded secret twice")
+        load_counts += 1
+        return k ** 2
+
+    c = TimeLimitedMaxSizeCache(load_secret_only_twice, one_day_ns, 2)
+    assert await c.lookup(3) == 9
+    assert await c.lookup(10) == 100
+    assert await c.lookup(3) == 9
+    assert await c.lookup(3) == 9
+    assert await c.lookup(10) == 100
+    try:
+        await c.lookup(4)
+    except ValueError:
+        assert load_counts == 2
+    else:
+        assert False
+
+
+async def test_lifetime():
+    load_counts = 0
+    async def load_secret(k: int):
+        nonlocal load_counts
+        load_counts += 1
+        return k ** 2
+
+    c = TimeLimitedMaxSizeCache(load_secret, one_second_ns, 3)
+    assert await c.lookup(3) == 9
+    assert load_counts == 1
+
+    assert await c.lookup(10) == 100
+    assert load_counts == 2
+
+    await asyncio.sleep(2)  # seconds
+    assert load_counts == 2
+
+    assert await c.lookup(3) == 9
+    assert load_counts == 3
+
+    assert await c.lookup(5) == 25
+    assert load_counts == 4
+
+    assert await c.lookup(10) == 100
+    assert load_counts == 5
+
+    # these two are a bit racy, but should be fine
```

Comment:
Perhaps just make the lifetime 4 seconds and sleep for 5 ealier? It shouldn't take _that_ long

Reply:
It is a good comment to leave but can you add a bit more noting the lifetime being so short? I missed that and it took me a minute to understand why it's racy